aws_lambda_events/src/dynamodb/mod.rs

```diff
@@ -1,11 +1,107 @@
-use chrono::{serde::ts_nanoseconds, DateTime, Utc};
+use chrono::{DateTime, Utc};
 use serde::{Deserialize, Serialize};
 use std::{collections::HashMap, fmt};
 
 pub mod attributes;
 use self::attributes::AttributeValue;
 
+pub mod ts_seconds_as_float {
+    use serde::{de, ser};
+    use std::fmt;
+
+    use chrono::offset::TimeZone;
+    use chrono::{DateTime, LocalResult, Utc};
+
+    enum SerdeError<V: fmt::Display, D: fmt::Display> {
+        NonExistent { timestamp: V },
+        Ambiguous { timestamp: V, min: D, max: D },
+    }
+
+    fn ne_timestamp<T: fmt::Display>(ts: T) -> SerdeError<T, u8> {
+        SerdeError::NonExistent::<T, u8> { timestamp: ts }
+    }
+
+    impl<V: fmt::Display, D: fmt::Display> fmt::Debug for SerdeError<V, D> {
+        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+            write!(f, "ChronoSerdeError({})", self)
+        }
+    }
+
+    impl<V: fmt::Display, D: fmt::Display> fmt::Display for SerdeError<V, D> {
+        fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+            match self {
+                &SerdeError::NonExistent { ref timestamp } => {
+                    write!(f, "value is not a legal timestamp: {}", timestamp)
+                }
+                &SerdeError::Ambiguous {
+                    ref timestamp,
+                    ref min,
+                    ref max,
+                } => write!(
+                    f,
+                    "value is an ambiguous timestamp: {}, could be either of {}, {}",
+                    timestamp, min, max
+                ),
+            }
+        }
+    }
+
+    fn serde_from<T, E, V>(me: LocalResult<T>, ts: &V) -> Result<T, E>
+    where
+        E: de::Error,
+        V: fmt::Display,
+        T: fmt::Display,
+    {
+        match me {
+            LocalResult::None => Err(E::custom(ne_timestamp(ts))),
+            LocalResult::Ambiguous(min, max) => Err(E::custom(SerdeError::Ambiguous {
+                timestamp: ts,
+                min,
+                max,
+            })),
+            LocalResult::Single(val) => Ok(val),
+        }
+    }
+
+    struct SecondsFloatTimestampVisitor;
+
+    /// Serialize a UTC datetime into an float number of seconds since the epoch
+    /// ```
+    pub fn serialize<S>(dt: &DateTime<Utc>, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: ser::Serializer,
+    {
+        serializer.serialize_i64(dt.timestamp_millis() / 1000)
+    }
+
+    /// Deserialize a `DateTime` from a float seconds timestamp
+    pub fn deserialize<'de, D>(d: D) -> Result<DateTime<Utc>, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        Ok(d.deserialize_f64(SecondsFloatTimestampVisitor)?)
+    }
+
+    impl<'de> de::Visitor<'de> for SecondsFloatTimestampVisitor {
+        type Value = DateTime<Utc>;
+
+        fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+            formatter.write_str("a unix timestamp as a float")
+        }
+
```

Comment:
@Miaxos you're missing these two functions in the visitor:
```suggestion
        fn visit_u64<E>(self, value: u64) -> Result<DateTime<Utc>, E>
        where E: de::Error,
        {
            serde_from(Utc.timestamp_opt(value as i64, 0), &value)
        }

        fn visit_i64<E>(self, value: i64) -> Result<DateTime<Utc>, E>
        where E: de::Error,
        {
            serde_from(Utc.timestamp_opt(value, 0), &value)
        }
```

I've tested your code in the playground, and it seems to work as expected with them.

- [Rust playground ](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=32190299df3622a731b5703953d0a8bc)
- [Go playground ](https://go.dev/play/p/kMXvw40KMeI) (for result comparison)